src/core/commit.rs

```diff
@@ -771,7 +771,9 @@ impl<J: Jet> CommitNode<J> {
     pub fn encode<W: io::Write>(&self, w: &mut BitWriter<W>) -> io::Result<usize> {
         let empty_witness = std::iter::repeat(Value::Unit);
         let program = self.finalize(empty_witness).expect("finalize");
-        program.encode(w)
+        let ret = program.encode(w)?;
```

Comment:
What about this line? Sorry for all the confusion

Reply:
Derp, you are correct. It was a false alarm. I removed the new commit.